.github/workflows/ci.yml

```diff
@@ -54,24 +58,41 @@ jobs:
           key: ${{ runner.os }}-sbt-cache-v2-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}
 
       - name: Check that workflows are up to date
-        run: sbt ++${{ matrix.scala }} githubWorkflowCheck
+        run: sbt 'project ${{ matrix.project }}' '++${{ matrix.scala }}' 'project /' githubWorkflowCheck
 
-      - name: Validate JVM
-        if: matrix.platform == 'jvm'
-        run: sbt ++${{ matrix.scala }} validateJVM
+      - name: fastOptJS
+        if: matrix.project == 'rootJS'
+        run: sbt 'project ${{ matrix.project }}' '++${{ matrix.scala }}' Test/fastOptJS
 
-      - name: Validate JS
-        if: matrix.platform == 'js'
-        run: sbt ++${{ matrix.scala }} validateJS
+      - name: nativeLink
+        if: matrix.project == 'rootNative'
+        run: sbt 'project ${{ matrix.project }}' '++${{ matrix.scala }}' Test/nativeLink
 
-      - name: Validate Native
-        if: matrix.platform == 'native'
-        run: sbt ++${{ matrix.scala }} validateNative
+      - name: Test
+        run: sbt 'project ${{ matrix.project }}' '++${{ matrix.scala }}' test
+
+      - name: Check binary compatibility
+        run: sbt 'project ${{ matrix.project }}' '++${{ matrix.scala }}' mimaReportBinaryIssues
+
+      - name: Generate API documentation
+        run: sbt 'project ${{ matrix.project }}' '++${{ matrix.scala }}' doc
+
+      - name: Make target directories
+        run: mkdir -p modules/deriving/.js/target modules/test/.jvm/target modules/test/.native/target target .js/target modules/typeable/.js/target modules/deriving/.jvm/target modules/typeable/.native/target .jvm/target .native/target local/.native/target local/.js/target modules/test/.js/target modules/typeable/.jvm/target local/.jvm/target modules/deriving/.native/target project/target
+
+      - name: Compress target directories
+        run: tar cf targets.tar modules/deriving/.js/target modules/test/.jvm/target modules/test/.native/target target .js/target modules/typeable/.js/target modules/deriving/.jvm/target modules/typeable/.native/target .jvm/target .native/target local/.native/target local/.js/target modules/test/.js/target modules/typeable/.jvm/target local/.jvm/target modules/deriving/.native/target project/target
+
+      - name: Upload target directories
+        uses: actions/upload-artifact@v2
+        with:
+          name: target-${{ matrix.os }}-${{ matrix.scala }}-${{ matrix.java }}
+          path: targets.tar
```

Comment:
Why is this running on PR?

Reply:
That's a good point. It's not harmful but probably not necessary. We can fix this upstream.